core/client/kt/requests.go

```diff
@@ -18,51 +18,45 @@ package kt
 import (
 	"fmt"
 
-	"github.com/google/keytransparency/core/crypto/signatures"
 	"github.com/google/keytransparency/core/crypto/vrf"
 	"github.com/google/keytransparency/core/mutator/entry"
 	"github.com/google/trillian/crypto/keyspb"
-
-	"github.com/google/trillian"
-
-	tpb "github.com/google/keytransparency/core/proto/keytransparency_v1_proto"
 )
 
-// CreateUpdateEntryRequest creates UpdateEntryRequest given GetEntryResponse,
-// user ID and a profile.
-func CreateUpdateEntryRequest(
-	trusted *trillian.SignedLogRoot, getResp *tpb.GetEntryResponse,
-	vrfPub vrf.PublicKey, domainID, userID, appID string, profileData []byte,
-	signers []signatures.Signer, authorizedKeys []*keyspb.PublicKey) (*tpb.UpdateEntryRequest, error) {
-	// Extract index from a prior GetEntry call.
-	index, err := vrfPub.ProofToHash(vrf.UniqueID(userID, appID), getResp.VrfProof)
+func (v *Verifier) index(vrfProof []byte, domainID, appID, userID string) ([]byte, error) {
+	uid := vrf.UniqueID(userID, appID)
+	index, err := v.vrf.ProofToHash(uid, vrfProof)
 	if err != nil {
-		return nil, fmt.Errorf("ProofToHash(): %v", err)
+		return nil, fmt.Errorf("vrf.ProofToHash(%v, %v): %v", appID, userID, err)
 	}
+	return index[:], nil
+}
 
-	oldLeaf := getResp.GetLeafProof().GetLeaf().GetLeafValue()
-	mutation := entry.NewMutation(index[:], domainID, appID, userID)
-	if err := mutation.SetPrevious(oldLeaf); err != nil {
-		return nil, fmt.Errorf("Error unmarshaling Entry from leaf proof: %v", err)
+// NewMutation creates a Mutation given the userID, desired state, and previous entry.
+func (v *Verifier) NewMutation(
+	domainID, appID, userID string,
```

Comment:
Having no stronger typing and a list of 3 different fooIDs makes me a bit nervous...  it'd be easy to transpose args by mistake in a call site and get some badly broken behaviour as a result...

Reply:
Yeah... Consistency between APIs should help, but I'm not sure what else to do.

Reply:
Well, the strong typing option is available:

type DomainID string
type AppID string
type UserID string

Do you think that would cause a lot of churn and annoyance?

Reply:
I kind of think it would cause more churn and type casting than really needed. 
If a call site gets it wrong all subsequent crypto proofs will fail, so there's an easy way to tell for sure if you got it right.